fe/fe-core/src/main/java/com/starrocks/catalog/HiveResource.java

```diff
@@ -61,4 +58,30 @@ protected void getProcNodeData(BaseProcResult result) {
     public String getHiveMetastoreURIs() {
         return metastoreURIs;
     }
+
+    /**
+     * <p>alter the resource properties.</p>
+     * <p>the user can not alter the property that the system does not support.
+     * currently , hive resource only support 'hive.metastore.uris' property to alter. </p>
+     *
+     * @param properties the properties that user uses to alter
+     * @throws DdlException
+     */
+    public void alterProperties(Map<String, String> properties) throws DdlException {
+        Preconditions.checkState(properties != null, "properties can not be null");
+
+        for (Map.Entry<String, String> entry : properties.entrySet()) {
+            String key = entry.getKey();
+            String value = entry.getValue();
+            if (HIVE_METASTORE_URIS.equals(key)) {
+                if (StringUtils.isBlank(value)) {
+                    throw new DdlException(HIVE_METASTORE_URIS + " can not be null");
+                }
+                validateMetastoreUris(value);
+                this.metastoreURIs = value;
```

Comment:
this member will be accessed concurrently by reader and writer.
Maybe should declare it with `volatile`